### PR TITLE
Dockerport

### PR DESCRIPTION
**Pull request by eliwaksbaum**
#327 everything seems to be working locally. now just need to integrate with openstack and make sure it works on compute canada

### CODE REVIEW

**Comment by eliwaksbaum** [comment]
Need a way to set DEBUG in the django settings. How can we detect if we're on compute canada or not?

**Comment by napulen** [comment]
Successfully launched the containers with `docker-compose`.

Static pages are working properly.

Trying to reach `/manuscripts` results in the following (after a long wait in the browser)

image

**Comment by napulen** [comment]
> Need a way to set DEBUG in the django settings. How can we detect if we're on compute canada or not?

We can simply set a "configurable" flag somehow to turn on `DEBUG`.

I'm thinking one can set an environment variable, then, if it exists, `docker` behaves differently.

```bash
CANTUS_ULTIMUS_DEBUG=True; docker-compose up
``` 

Or something similar.

**Comment by napulen** [comment]
Maybe relevant with `postgres` issue: https://stackoverflow.com/questions/51750715/could-not-translate-host-name-db-to-address-using-postgres-docker-compose-and

**Comment by eliwaksbaum** [comment]
> Maybe relevant with `postgres` issue: https://stackoverflow.com/questions/51750715/could-not-translate-host-name-db-to-address-using-postgres-docker-compose-and

Yeah, I think `restart:always` is all over the place in rodan's dockerfiles, that could be useful

**Comment by napulen** [comment]
@eliwaksbaum All of the recent commits are related to unit tests, correct?

We need to split those into a separate PR. Given that I am the bottleneck (sorry!) for the Dockerport PR getting merged, please implement the unit test commits over the vagrant (`main`) branch. We will merge them in this order:

- Unit tests PR merges into `main`, on top of the recent features that @dchiller implemented for the mapping function (thanks, Dylan!)
- Dockerport will be a bigger PR that will merge after @napulen and @dchiller have successfully worked with the docker version of Cantus Ultimus. I anticipate a few more commits coming for Dockerport before merging (e.g., the weird `postgres` stuff)

**Comment by napulen** [comment]
@eliwaksbaum in order to rescue all of your work, you can use a combination of `git revert` and `git cherry-pick` to take the last few commits off this branch and push them into another one branching from `main`.

You can call the new branch `unit_tests_2022` or similar. And that branch is what we merge first.

**Comment by eliwaksbaum** [comment]
@napulen can you try copying [wait-for-it.sh](https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh) into the app directory, and then changing app's command in docker-compose.yaml to `command: ["/code/wait-for-it.sh", "postgres:5432", "--", "/code/django-config.sh"]`

**Comment by napulen** [comment]
@eliwaksbaum yes, I'll try this tomorrow (thesis day today!)

**Comment by napulen** [comment]
Confirming that `wait-for-it.sh` works for me.

**Review by napulen — CHANGES_REQUESTED** [review]
Ok, we are almost ready to merge!

- Something's up with the `cantus-staticpages` submodule.
- Some other minor comments
~- I might not be able to correct the merge conflict myself, can you take a look? The conflicting file is `etc/provision/setup-privileged.sh`~

image

> Nevermind, I do have to merge the conflict myself, I just need to do it outside the web GUI.

As soon as we get those resolved, we can merge

File: .env
Comment (eliwaksbaum) [reply .env:1]: not sure what you mean. what I was using that for is so that all the images have a constant name. images are named {PROJECT_NAME}_app etc. and at one point we need to reference one. This makes sure that reference is always valid, even if someone changes the name of the folder

File: app/django-config.sh
Comment (eliwaksbaum) [reply app/django-config.sh:1]: the problem is that django's migration stuff relies on talking to the postgres container, which can't be done during the build process. key.sh could be a RUN directive, yes

File: app/cantus-staticpages/LICENSE
Comment (eliwaksbaum) [reply app/cantus-staticpages/LICENSE:1]: had to wrestle with git for a while but i think i got it

File: .env
Comment (napulen) [reply .env:1]: Gotcha. I was just asking because the name `.env` has a special meaning sometimes (e.g., python virtual environments) and wasn't sure if we needed that file committed as part of the repo. Based on what you said, we do need it.

File: app/cantus-staticpages/LICENSE
Comment (napulen) [reply app/cantus-staticpages/LICENSE:1]: Cool! Any new commits out of that?

**Review by napulen — APPROVED** [review]

**Comment by napulen** [comment]
aaand merged. Great work, @eliwaksbaum

**Pull request merged**
